opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -248,8 +250,8 @@ namespace cpgrid
 
 
     /// Read the Eclipse grid format ('.grdecl').
-    void CpGridData::processEclipseFormat(const grdecl& input_data, const NNCMaps& nnc, double z_tolerance, bool remove_ij_boundary, bool turn_normals,
-                                          const std::unordered_map<size_t, double>& aquifer_cell_volumes)
+    void CpGridData::processEclipseFormat(const grdecl& input_data, Opm::EclipseState* ecl_state, const Opm::Deck* deck,
+                                          NNCMaps& nnc, double z_tolerance, bool remove_ij_boundary, bool turn_normals)
```

Comment:
I didn't notice this before.  Strictly speaking, this particular function is not protected by `HAVE_ECL_INPUT` so it was originally supposed to be callable without types from opm-common.  I'd be happy to remove that restriction, but maybe we should discuss this topic a little more before we merge the PR.

Reply:
Sure. I am not that familiar with grid related stuff. 

In my own opinion, the grid processing (getting the final ACTNUM, we should not drop any information before determining the final ACTNUM, so a cell is free to change from inactive to active.) should be done in the early stage of the parsing, so all the numerical aquifers stuff can stay in opm-common.  

The current way is intruding because we do the processing (can remove more cells) after parsing, while parsing needs processed grid and parsing information also intervene the grid processing.